File: KBDTypeServer/Application/DTOs/AddressDtos/AddressDto.cs
using System.ComponentModel.DataAnnotations;

namespace KBDTypeServer.Application.DTOs.AddressDtos
{
    public class AddressDto
    {
        [Required]
        public int Id { get; set; }
        [Required]
        public int UserId { get; set; }
        [Required]
        public string Region { get; set; }
        [Required]
        public string City { get; set; }
        [Required]
        public string Street { get; set; }

        public string? Apartment { get; set; }
        [Required]
        public string Building { get; set; }
        [Required]
        public string PostalCode { get; set; }

    }
}

File: KBDTypeFront/src/dto/address/AddressDto.ts
export class AddressDto {
  id: number = 0;
  userId: number = 0;
  region: string = "";
  city: string = "";
  street: string = "";
  building: string = "";
  apartment?: string | null = null;
  postalCode: string = "";

  constructor(init?: Partial<AddressDto>) {
    Object.assign(this, init);
  }
}